tests/precision_checking.rs

```diff
@@ -0,0 +1,130 @@
+// these tests check the asserts that are only present in debug configurations
+// so they won't pass in release mode
+#![cfg(debug_assertions)]
```

Comment:
is this required? Do you typically find yourself running tests in release mode?

Reply:
I don't do it usually aside from cases where it might be too slow in debug.

It's still technically possible (`cargo test --release`), so I decided to guard against that just in case™.